lib/ansible/modules/cloud/vmware/vmware_object_rename.py

```diff
@@ -0,0 +1,291 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2019, Ansible Project
+# Copyright: (c) 2019, Abhijeet Kasurde <akasurde@redhat.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = r'''
+---
+module: vmware_object_rename
+short_description: Renames VMware objects
+description:
+- This module can be used to rename VMware objects.
+- All variables and VMware object names are case sensitive.
+- Renaming Host and Network is not supported by VMware APIs.
+version_added: '2.10'
+author:
+- Abhijeet Kasurde (@Akasurde)
+notes:
+- Tested on vSphere 6.5
+requirements:
+- python >= 2.7
+- PyVmomi
+- vSphere Automation SDK
+options:
+    object_type:
+      description:
+      - Type of object to work with.
+      - Valid options are Cluster, ClusterComputeResource, Datacenter, Datastore, Folder, ResourcePool, VM or VirtualMachine.
```

Comment:
The valid options can be mentioned under `choices` key like mentioned here - https://docs.ansible.com/ansible/2.3/dev_guide/developing_modules_documenting.html

Reply:
I am aware that `choices` key available but Object types are dynamically populated from VMware APIs since I didn't want to hard code the values. This line is for user information only.

So the idea is if VMware adds a new object in the future, I don't need to update this block code. Just documentation update will suffice in that case.